src/inventory.cpp

```diff
@@ -246,6 +246,19 @@ std::string ItemStack::getItemString() const
 	return os.str();
 }
 
+std::string ItemStack::getDescription(bool append_itemname,
+		IItemDefManager *itemdef) const
+{
+	std::string desc = metadata.getString("description");
+	if (desc.empty())
+		desc = getDefinition(itemdef).description;
+	if (desc.empty())
+		return name;
+	if (append_itemname)
+		desc += "\n[" + name + "]";
```

Comment:
I'd prefer if this client-only feature would added in `guiFormSpecMenu.cpp`. If you'd like to adapt that, it's also possible to set an input pointer to `true` just like `getNode` does for `is_valid_position`.

Reply:
Moved it, but now the tooltip could have the itemname twice if there's no description. I think that's ok since `tooltip_append_itemname` is mostly for developers and items in the inventory will normally have a description.

Reply:
The new implementation isn't really better, also the behavior is different.
Might just be better to leave it with the parameter.